core_contracts/dex/dex.py

```diff
@@ -1381,15 +1464,25 @@ def totalBalnAt(self, _id: int, _snapshot_id: int) -> int:
 
     @external(readonly=True)
     def getTotalValue(self, _name: str, _snapshot_id: int) -> int:
-        # return self.totalSupplyAt(self._named_markets[_name], _snapshot_id)
-        return self.totalSupply(self._named_markets[_name])
+        twa = self._time_weighted_lp.get()
+        if not twa:
+            return self.totalSupply(self._named_markets[_name])
+        else:
+            return self.totalSupplyAt(self._named_markets[_name], _snapshot_id)
```

Comment:
It would be good to remove the code duplicate like below:
```suggestion
        twa = self._time_weighted_lp.get()
        return self.getTotalValueOptional(_name, _snapshot_id, _twa=twa)
```